HIP-0003.md

```diff
@@ -0,0 +1,95 @@
+This provider will respond to the request with the signed challenge sent by the Service.
+
+To be a provider, the Provider needs to register and respond for the [custom protocol](https://html.spec.whatwg.org/#web+-scheme-prefix) `web+hns`.
+
+This protocol will be used by the Service to send authentication challenges.
+
+Assuming that the community will provide different solutions and implementation options, the User will choose one Provider and register it to respond to the custom protocol.
+
+It is outside the context of this HIP to define how the Provider interacts with the User or how the public/private keys should be generated.
+
+This also means that, once the User chooses a Provider and is onboarded, the User won't be able to switch to another provider without replacing its public key. In other words: the User cannot authenticate itself with any Provider, but only with the Provider he/she used to generate the public key.
+
+### Exposing the public key
+
+The public key should be exposed via HTTPS request to `.well-known/authentication`. For the domain `example` the full URL should be `https://example/.well-known/authentication`.
+
+This url should use a SSL certificate validate with [DANE](https://tools.ietf.org/html/rfc6698) or by a Certificate authority.
+
+### Authentication request
+
+Any Service can create an authentication request. It's a basic `GET` request to `web+hns://authentication/?challenge=<CHALLENGE>&callback=<CALLBACK>`.
+
+| Key            | Type                 | Options                                                                                                                          |
+| -------------- | -------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| web+hns://     | protocol             | Fixed.                                                                                                                           |
+| authentication | Reserved URL keyword | Fixed.                                                                                                                           |
+| challenge      | querystring          | A 32-characters long string. Hex.                                                                                                |
+| callback       | querystring          | An encoded callback URL. It will be (async) called by the provider with the challenge response. It must be a secure URL (HTTPS). |
+
+### Authentication response
+
+It's a Provider's decision on how to identify and authenticate the user with the Provider.
+
+Once the provider identifies the User, it can sign the challenge using the User's private key and call the Service using the provided callback URL in a `GET` request.
```

Comment:
> Once the Provider identifies the User, it can sign the challenge using the User's private key and call the Service using the provided callback URL in a `GET` request.